subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -858,10 +858,11 @@ void ticker_worker(void *param)
 			if ((ticker->must_expire == 0U) ||
 			    (ticker->lazy_periodic >= ticker->lazy_current) ||
 			    TICKER_RESCHEDULE_PENDING(ticker)) {
-				/* Not a must-expire case, this is programmed
+				/* Not a must-expire node or this is periodic
 				 * latency or pending re-schedule. Skip this
-				 * ticker node.
+				 * ticker node. Mark it as elapsed.
 				 */
+				ticker->ack--;
```

Comment:
Thinking aloud,
1. If not `must_expire`, mark ticker instance as expired without resetting the `lazy_current`. This is desired behavior, the ticker will be scheduled at next interval period, with incremented `lazy_current`.
2. If the incremented `lazy_current` is less than `lazy_periodic`, the event callback will not be called even if `must_expire` is true. This is correct.
3. If `TICKER_RESCHEDULE_PENDING(ticker)` then mark it as expired without resetting the `lazy_current`. Would this not cause incorrect `lazy_current` ? (I think only advertiser uses this which does not bother of any incorrect `lazy` value, and hence we dont notice issues upstream?)

We need to work in consideration of changes done in https://github.com/zephyrproject-rtos/zephyr/pull/23815

and attention to this change: https://github.com/zephyrproject-rtos/zephyr/commit/b263c2860e7753beb72c70eca4e5de7fc334aa23

That is, the change/fix to reset the ticker state is inline with the fact that ticker state is put into scheduling requested state down in the code in the `ticker_job_worker_bh` function.

Now getting back to `ticker->ack--;`, a decision on whether we want `ticker->lazy_current++;` under the 3 cases (1,2,3 above) will solve the incorrect `lazy_current` that I assume causes the link loss.

Case 1 and case 2, is correctly handled. Case 3, is faulty (correct me with your independent understanding).

Finally, I have not disapproval to the addition of `ticker->ack--;` as it will put the ticker state in the correct state for it to be handled in `ticker_job_worker_bh`

Reply:
> 3. If `TICKER_RESCHEDULE_PENDING(ticker)` then mark it as expired with the incremented `lazy_current`. Would this not cause incorrect `lazy_current` ? (I think only advertiser uses this which does not bother of any incorrect `lazy` value, and  hence we dont notice issues upstream?)

I understand your concern for case 3. However, if I understand your point correctly, I believe this is handled in line 1828 of `ticker_job_reschedule_in_window`, where `lazy_current` is always decremented for re-scheduled nodes.

Also, I have a comprehensive ztest-suite which indicates correct functionality of the `slot_window` feature. This of course could be flawed, so please elaborate on your concern. 

I have taken the recent changes you have mentioned into consideration (and testing), and as far as I can tell, they all work well together, event if my change partly reverts some of the logic in #23815.

Reply:
Obviously, I missed the `lazy_current--` here https://github.com/zephyrproject-rtos/zephyr/blame/master/subsys/bluetooth/controller/ticker/ticker.c#L1824

Please add a similar comment in `ticker_worker` so that I dont misunderstand again in the future.